### DISCUSSION TITLE
GitHub Action registering new ECS task definition

### DISCUSSION THREAD

**Discussion by tsposato** [community/community#25308] [Actions]
<p>Hi,</p>
<p>I am trying to migrate from in-house Jenkins to GitHub actions for our CI-CD pipelines.<br>
I have created an ‘action.yml’ file in the ‘.github/workflows/’ folder.<br>
The job builds 2 separate docker containers, tagging them both twice and then pushing to AWS ECR which all works fine.</p>
<p>The error I am getting is from the below:</p>
<pre><code class="lang-auto"> Deploy_to_ECS:
    name: Deploy to ECS
    runs-on: ubuntu-latest
    needs: [Build_Push_Docker_Images]
    steps:
      - uses: actions/checkout@v2

      - name: Get short SHA
        id: slug
        run: echo "::set-output name=sha7::$(echo ${GITHUB_SHA} | cut -c1-7)"

      - name: Render Amazon ECS task definition for php-fpm container
        id: render-php-container
        uses: aws-actions/amazon-ecs-render-task-definition@v1
        with:
          task-definition: task-definition.json
          container-name: "php-fpm"
          image: "${{ env.AWS_ACCOUNT_ID }}.dkr.ecr.${{ env.AWS_DEFAULT_REGION }}.amazonaws.com/${{ env.IMAGEPHPFPM }}:${{ steps.slug.outputs.sha7 }}"
</code></pre>
<p>The error I get is: /home/runner/work/the-first/the-first/task-definition.json: Unexpected token � in JSON at position 0<br>
The file is valid json, I have checked over and over again and I am at a loss on how to proceed. Using normal aws cmd line tools with this task-definition.json file successfully registers a new task definition in ECS.<br>
Debug logs don’t shed any more light on the situation either.<br>
Any help would be much appreciated.</p>
<p>Thanks,<br>
Terry</p>

**Comment by otter-computer** [comment]
<p>Hi <a class="mention" href="/tsposato">@tsposato</a>, welcome to the GitHub Support Community!</p>
<p>I’m wondering if this is an issue with the <code>aws-actions/amazon-ecs-render-task-definition</code> action itself rather than an issue with your workflow? If you’re saying this same JSON file works when using it from the aws cli but doesn’t when you use this action it sounds like there may be a problem with this action consuming the JSON file.</p>
<p>It may be worth opening an issue for this inside the repository for this action. They’ll likely have a better idea of what’s going wrong and whether it is something wrong with the action itself or something wrong with your workflow.</p>
<aside class="onebox githubfolder">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">
      <a href="https://github.com/aws-actions/amazon-ecs-render-task-definition/issues" target="_blank" rel="noopener">github.com</a>
  </header>
  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:210/210;"><img src="https://avatars3.githubusercontent.com/u/57468012?s=400&amp;amp;v=4" class="thumbnail" width="210" height="210"></div>

<h3><a href="https://github.com/aws-actions/amazon-ecs-render-task-definition/issues" target="_blank" rel="noopener">aws-actions/amazon-ecs-render-task-definition</a></h3>

<p><a href="https://github.com/aws-actions/amazon-ecs-render-task-definition/issues" target="_blank" rel="noopener">s</a></p>

  <p><span class="label1">Inserts a container image URI into an Amazon ECS task definition JSON file.</span></p>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

**Comment by tsposato** [comment]
<p>Hi <a class="mention" href="/thomasshaped">@thomasshaped</a>, thanks!<br>
I feel a bit silly now but it seems like my IDE is doing something strange with the JSON file and causing this issue. When I use the aws cli on the JSON file in my github repo I do indeed get the error that it is receiving invalid JSON. Which is quite strange. I’m still trying to figure out why. If I copy and paste it into an external JSON validator it comes up as valid JSON. I guess I’ll continue to try and figure out why.</p>

**Comment by tsposato** [comment, marked as answer]
<p>Ok, the problem was the encoding of the file was not ‘UTF-8’. Once I changed the encoding of the file I was able to register a Task Definition successfully with the aws cli and with the github action <code>aws-actions/amazon-ecs-render-task-definition</code>.</p>
<p>Thanks.</p>

**Comment by matbeedotcom** [comment]
<p>Excellent, tsposato, this fixed my error as well. Thank you.</p>
